src/main/java/com/JessZuchowski/codefellowship/controllers/UserController.java

```diff
@@ -0,0 +1,133 @@
+import com.JessZuchowski.codefellowship.database.ApplicationUser;
+import com.JessZuchowski.codefellowship.database.UserRepository;
+import net.bytebuddy.asm.Advice;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.view.RedirectView;
+
+import java.util.Optional;
+
+@Controller
+public class UserController {
+
+    @Autowired
+    UserRepository repo;
+
+    @Autowired
+    PasswordEncoder encoder;
+
+    @GetMapping
+    public String getIndex(@AuthenticationPrincipal ApplicationUser user) {
+        if (user != null) {
+            return "redirect:/myprofile/" + user.getId();
+        }
+        return "index";
+    }
+
+    @GetMapping("/landing")
+    public String getLanding() {
+        return "index";
+    }
+
+    @GetMapping("/signup")
+    public String getSignup(
+            @AuthenticationPrincipal ApplicationUser user
+    ) {
+        if (user != null) {
+
+        }
+        return "signup";
+    }
+//    POST, save to DB
+    @PostMapping("/signup")
+    public RedirectView postSignup(
+            @RequestParam String username,
+            @RequestParam String password,
+            @RequestParam String firstName,
+            @RequestParam String lastName,
+            @RequestParam String dateOfBirth,
+            @RequestParam String bio
+    ) {
+        ApplicationUser user = new ApplicationUser();
+        user.setUsername(username);
+        user.setPassword(encoder.encode(password));
+        user.setFirstName(firstName);
+        user.setLastName(lastName);
+        user.setDateOfBirth(dateOfBirth);
+        user.setBio(bio);
+
+        repo.save(user);
+
+        //to auto login
+        UsernamePasswordAuthenticationToken token = new UsernamePasswordAuthenticationToken(
+                user,
+                null,
+                user.getAuthorities());
+        SecurityContextHolder.getContext().setAuthentication(token);
+
+        return new RedirectView("/myprofile/" + user.getId());
+    }
+
+//    login
+    @GetMapping("/login")
+    public String getLogin(
+            @RequestParam(required = false, defaultValue = "false") boolean isErrored,
+            Model model
+    ) {
+        model.addAttribute("errored", isErrored);
+        return "login";
+    }
+
+    @GetMapping("/login-error")
+    @ResponseBody
+    public String getLoginError(Model model) {

```

Comment:
This method doesn't make sense in the context. If you hit this route, the only thing displayed is the string"login".